### PR TITLE
Fixed low energy GFLASH

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
This PR intend to fix the problem of low energy GFLASH method (see https://indico.cern.ch/event/1242712/).

because this feature is not enabled by default, no change in any WF is expected.

#### PR validation:
Private using single gamma and BT2006 setups.

Will be backported to 13_0

### CODE REVIEW

**Comment by civanch** [comment]
type bug-fix

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
please test

timeout again

**Comment by civanch** [comment]
+1

timeout cannot be connected to this PR because fixed feature is not used in any WF.

**Comment by perrotta** [comment]
@civanch is your private validation verifying that the issues reported in https://indico.cern.ch/event/1242712/contributions/5222797/attachments/2579281/4448206/PdmV_electrons_20012023.pdf will can get solved by this PR? Could you link a few plots to the PR description?

**Comment by rappoccio** [comment]
@civanch just a quick ping for a plot from the validation here so we can merge this.

**Comment by civanch** [comment]
@perrotta , @rappoccio , I am adding some plots. The first from test beam:
image

**Comment by perrotta** [comment]
Thank you @civanch 

In the PR description the issue that this PR is supposed to fix was the overestimated energy for low-energy photons and electrions, i,e.
PhotonE
ElectronE

The reduction of the `scaleFactor`  by some 2-5% which is implemented at [L19](https://github.com/cms-sw/cmssw/pull/40819/files#diff-860a4ef4d927d366a1b0bd6915c1046284a6eb81db367ffab11bcdd9d4c04cc6R19) I imagine can go in the right direction, and your comparisons with some test beam results may suggest that this is now actually the case (even though a reproduction of the same plots linked to the PR description would have better shown that the fix really propagate to the previously failing comparisons.

For the photons the validations linked to the PR description also reported about some reduced distance between the (primary) photon and the PF neutral hadron candidate
PhotonDr
Is this what the update/fix in this PR of the `LowEnergyFastSimModel::ModelTrigger` method is supposed to cure?

Given the correct order of magnitude of the scale factor reduction implemented here I think this PR can be merged anyhow, even if some demonstration usign similar comparisons as the ones that were failing could have been preferred. Still probably some justification (and/or explanation) of the update (fix?) to `ModelTrigger` should be better added here, for documentation

**Comment by rappoccio** [comment]
@civanch pinging on this

**Comment by civanch** [comment]
@perrotta , @rappoccio , fixes in the lines after 65 are critical: the correct method is proposed to identify if the primary particle producing electromagnetic shower is gamma, e+, or e-. In the previous version of the code there were incorrect identification of very primary particle. In the update version GFLASH does not applied to low-energy e+- if they are produced in such showers. This done for purpose not affect gamma, e+- identification inside ECAL, so the method may be used without re-calibration.

The change of the scale parameters is introduced in order to fix visible energy from initial gamma, which are applied to low-energy e+- coming from hadrons, not from electromagnetic showers.

In summary, these fix of the code is correct and it should be merged. Scale factor restoring scale for gamma.  

I was late with response, because the method itself is required further evaluation - I am not 100% satisfied by the results but this PR is correct.

**Comment by perrotta** [comment]
+1
- Thank yuo @civanch 
- Looking forward another validation as in https://indico.cern.ch/event/1242712/ can be performed after the merge of this PR, to verify if there were improvements for it
